### DISCUSSION TITLE
Pre-requisites of Patrick's course on Smart Contract Security

### DISCUSSION THREAD

**Discussion by 5h3rl0ckH0lm3s** [Cyfrin/foundry-full-course-cu#1787] [Q&A]
I am interested in learning Smart Contract Security & Auditing but not in making projects in foundry. I have completed till Lesson 8 of the foundry-full-course and now I am thinking to skip the 9th lesson that is 'Smart Contract Lottery' and move on from 10th lesson. My question is from the people who have completed Lesson 9 and also the Smart Contract Security Course by Patrick, that do I need to complete lesson 9 for doing the 'Smart Contract Security & Auditing' course in future. 
Are there some important concepts in Lesson 9 that I need to learn before moving on ? If there are, please list them.

**Comment by EngrPips** [comment]
Hello @5h3rl0ckH0lm3s, You can skip the `Smart Contract Lottery` if you think you are pressed for time, but it would be great if you did it because it introduces and teaches some new concepts, such as chainlink `VRF` and `Keepers`. I don't see any harm in skipping it.

**Reply by 5h3rl0ckH0lm3s** [reply]
@EngrPips, thank you very much for answering. Can you please list other new concepts also that are introduced in the 9th Lesson, so that I can be aware about them before moving on ?

**Reply by EngrPips** [reply]
Aside from those two concepts, what I could remember again is that the `smart contract Lottery` section teaches how to listen to events in a test and extract data from emitted events. Aside from all of these, it is just writing contracts, scripts, and tests like we have been doing all along as far as I can remember

**Reply by 5h3rl0ckH0lm3s** [reply]
Ok, thank you very much @EngrPips.

**Reply by EngrPips** [reply]
Anytime, friend. I am glad I could help.

**Reply by 5h3rl0ckH0lm3s** [reply]
@EngrPips can you do me one more favour please ?
As I said, I wanted to do smart contract security and Auditing, so do I also need to complete the Lesson 12 on Foundry DeFi of Patrick's foundry-full-course to do that ?

**Reply by EngrPips** [reply]
Completing the whole of `foundry-full-course` would make the Auditing course really easy but if you are pressed for time you can jump to the auditing course with a good understanding  of solidity and Defi you shouldn't have a problem.
